### DISCUSSION TITLE
Expand EEPROM memory for more features? "Egzumer +"...?

### DISCUSSION THREAD

**Discussion by sztuk5** [egzumer/uv-k5-firmware-custom#377] [Ideas]
Hi, I was thinking... If there is no enough memory to fit all the functions, then why not to expand it? Especially in such cheap radio.

I found that **BL24C128A** should work (and on Aliexpress cost about **10-20zł/$3-5 per 10 pcs**) in place of **BL24C64A** that is in Qunashnegs right now. Unfortunately I don't have enough knowledge to do it myself. With AI chat, I get to know that there are some parameters that have to be changed for this to work properly (e.g. in EEPROM files). Also, that other components in this radio have to be compatible with higher capacity EEPROM and that they probably are. 

All of that should be checked by someone who have knowledge in this field, but it seems that it can be done with relatively small amount  of effort. This might be like maybe second version of Egzummer ("Egzumer +" maybe ;p ) for higher capacity EEPROM.

Since here is where my knowledge ends, I thought that maybe starting a discussion can push this idea forward. I wonder how many cool features can this radio fit having twice as much memory :D

**Comment by JuantAldea** [comment]
This is not trivial. The EEPROM is used for storing persistent data, not the program. Using the EEPROM to store bits of FW would required implementing all the infrastructure to load, execute, and unload such applets.

**Reply by sztuk5** [reply]
So where is this mod stored? I heard or read somewhere that there is 64 Kbits of storage in this radio which I thought corresponds to BL24C64A chip as it also have 64 Kbits of memory, and so I assumed that it's used to store this mod.

**Reply by JuantAldea** [reply]
In the FLASH memory of the microprocessor (Cortex-M0, DP32G030)

**Reply by JuantAldea** [reply]
And the RAM is 16KB, which is where the code to be executed would have to be copied.

**Reply by sztuk5** [reply]
Thank You for replaying and explanation. That's a shame :/. I genuinely had hope for a moment that it will be easy. I really miss having knowledge like that about electronic circuits. I also don't think that if it's all integrated, then it will be an option to expand it in any way. Additionally I searched  for compatible replacement processor with larger memory a little, but I didn't find any, so I guess topic ends here. My only hope now is that maybe compression will get better.

**Comment by ErikS-web** [comment]
I don't think this is an "Issue on actual firmware".

And should be continued under Discussions.

But I do like this hobby-experiments and cannot wait to desolder the eeprom with hot-air and replace with bigger one ;-)
